.ci/no_old_refs.sh

```diff
@@ -5,27 +5,66 @@ set -e
 # Script requires GITHUB_TOKEN env variable
 MENTIONED_ISSUES=/tmp/mentioned_issues
 CLOSED_ISSUES=/tmp/failed_issues
+LINKED_ISSUES_FORMATTED=/tmp/linked_issues
+
+# Linked issues that are mentioned in the code
+LINKED_ISSUES_MENTIONED=/tmp/linked_issues_mentioned
 API_GITHUB_PREFIX="https://api.github.com/repos"
 GITHUB_HOST="https://github.com"
+CHECKSTYLE_ISSUE_PREFIX="https:\/\/github.com\/checkstyle\/checkstyle\/issues\/"
+MAIN_REPO="checkstyle/checkstyle"
+DEFAULT_BRANCH="master"
+
+# These are modified when event is of type pull_request
+if [ ! -z "$PR_HEAD_REPO_NAME" ]; then
+  MAIN_REPO=$PR_HEAD_REPO_NAME
+  DEFAULT_BRANCH=$GITHUB_HEAD_REF
+fi
 
 # collect issues where full link is used
-grep -IPohr "(after|[Tt]il[l]?) $GITHUB_HOST/[\w.-]+/[\w.-]+/issues/\d{1,5}" . \
-  | sed -e 's/.*github.com\///' >> $MENTIONED_ISSUES
+grep -IPonr "(after|[Tt]il[l]?) $GITHUB_HOST/[\w.-]+/[\w.-]+/issues/\d{1,5}" . \
+  | sed -e 's/:[a-zA-Z].*github.com\//:/' >> $MENTIONED_ISSUES
 
 # collect checkstyle issues where only hash sign is used
-grep -IPohr "[Tt]il[l]? #\d{1,5}" . \
-  | sed -e 's/.*#/checkstyle\/checkstyle\/issues\//' >> $MENTIONED_ISSUES
+grep -IPonr "[Tt]il[l]? #\d{1,5}" . \
+  | sed -e 's/:[a-zA-Z].*#/:checkstyle\/checkstyle\/issues\//' >> $MENTIONED_ISSUES
+
+# $LINKED_ISSUES need formatting before the are used
+if [ ! -z "$LINKED_ISSUES" ]; then
+  echo $LINKED_ISSUES | sed -e 's/,/\n/g' >> $LINKED_ISSUES_FORMATTED
+  sed -i "s/^/$CHECKSTYLE_ISSUE_PREFIX/" $LINKED_ISSUES_FORMATTED
+fi
+
+for MENTIONED_ISSUES_LINES in $(cat $MENTIONED_ISSUES); do
+  issue=${MENTIONED_ISSUES_LINES#*[0-9]:}
+
+  location=${MENTIONED_ISSUES_LINES%:[0-9]*}
+  location=${location:2}
+
+  line_number=${MENTIONED_ISSUES_LINES#*:}
+  line_number=${line_number%:*}
+
+  link="$GITHUB_HOST/$MAIN_REPO/blob/$DEFAULT_BRANCH/$location#L$line_number"
 
-for issue in $(sort -u $MENTIONED_ISSUES); do
   STATE=$(curl -s -H "Authorization: token $GITHUB_TOKEN" "$API_GITHUB_PREFIX/$issue" \
    | jq '.state' | xargs)
   if [ "$STATE" = "closed" ]; then
-    echo "$GITHUB_HOST/$issue" >> $CLOSED_ISSUES
+    echo "$link" >> $CLOSED_ISSUES
+  elif [ ! -z "$LINKED_ISSUES" ]; then
+    for linked_issue in $(cat $LINKED_ISSUES_FORMATTED); do
+      if [ "$linked_issue" = "$GITHUB_HOST/$issue" ]; then
+        echo "$link" >> $LINKED_ISSUES_MENTIONED
+      fi
+    done
   fi
 done
 
 if [ -f "$CLOSED_ISSUES" ]; then
-    echo "Following issues are mentioned in code to do something after they are closed:"
+    echo -ne "\nFollowing issues are mentioned in code to do something after they are closed:\n\n"
     cat $CLOSED_ISSUES
+    if [ -f "$LINKED_ISSUES_MENTIONED" ]; then
+      echo -ne "\nFollowing issues are linked to the PR and are also mentioned in the code:\n\n"
```

Comment:
We need to hint user that link should be removed.

Reply:
Done.